docs/zh-CN/administrator-guide/materialized_view.md

```diff
@@ -0,0 +1,490 @@
+	|      tabletRatio=10/10                                                                                                                                            |
+	|      tabletList=22084,22086,22088,22090,22092,22094,22096,22098,22100,22102                                                                                       |
+	|      cardinality=0                                                                                                                                                |
+	|      avgRowSize=0.0                                                                                                                                               |
+	|      numNodes=1                                                                                                                                                   |
+	+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+	45 rows in set (0.030 sec)
+	```
+	
+	在 EXPLAIN 的结果中，首先可以看到 OlapScanNode 的 rollup 属性值为 advertiser_uv。也就是说，查询会直接扫描物化视图的数据。说明匹配成功。
+	
+	其次对于 `user_id` 字段求 `count(distinct)` 被改写为求 `bitmap_union_count(to_bitmap)`。也就是通过 bitmap 的方式来达到精确去重的效果。
+	
+
+## 最佳实践3
+
+业务场景：匹配更丰富的前缀索引
+
+用户的原始表有 （k1, k2, k3） 三列。其中 k1, k2 为前缀索引列。这时候如果用户查询条件中包含 `where k1=1 and k2=2` 就能通过索引加速查询。
+
+但是有些情况下，用户的过滤条件无法匹配到前缀索引，比如 `where k3=3`。则无法通过索引提升查询速度。
+
+创建以 k3 作为第一列的物化视图就可以解决这个问题。
+
+1. 创建物化视图
+
+	```
+	CREATE MATERIALIZED VIEW mv_1 as SELECT k3, k2, k1 FROM tableA ORDER BY k3;
+	```
+	
+	通过上面语法创建完成后，物化视图中既保留了完整的明细数据，且物化视图的前缀索引为 k3 列。表结构如下：
+	
+	```
+	MySQL [test]> desc tableA all;
+	+-----------+---------------+-------+------+------+-------+---------+-------+
+	| IndexName | IndexKeysType | Field | Type | Null | Key   | Default | Extra |
+	+-----------+---------------+-------+------+------+-------+---------+-------+
+	| tableA    | DUP_KEYS      | k1    | INT  | Yes  | true  | NULL    |       |
+	|           |               | k2    | INT  | Yes  | true  | NULL    |       |
+	|           |               | k3    | INT  | Yes  | true  | NULL    |       |
+	|           |               |       |      |      |       |         |       |
+	| mv_1      | DUP_KEYS      | k3    | INT  | Yes  | true  | NULL    |       |
+	|           |               | k2    | INT  | Yes  | false | NULL    | NONE  |
+	|           |               | k1    | INT  | Yes  | false | NULL    | NONE  |
+	+-----------+---------------+-------+------+------+-------+---------+-------+
+	```
+	
+2. 查询匹配
+
+	这时候如果用户的查询存在 k3 列的过滤条件是，比如：
+	
+	```
+	select k1, k2, k3 from table A where k3=3;
+	```
+	
+	这时候查询就会直接从刚才创建的 mv_1 物化视图中读取数据。物化视图对 k3 是存在前缀索引的，查询效率也会提升。
+
+
+## 局限性
+
+1. 物化视图的聚合函数的参数不支持表达式仅支持单列，比如： sum(a+b)不支持。
+2. 如果删除语句的条件列，在物化视图中不存在，则不能进行删除操作。如果一定要删除数据，则需要先将物化视图删除，然后方可删除数据。
+3. 单表上过多的物化视图会影响导入的效率：导入数据时，物化视图和 base 表数据是同步更新的，如果一张表的物化视图表超过10张，则有可能导致导入速度很慢。这就像单次导入需要同时导入10张表数据是一样的。
+
```

Comment:
4. the same column with two aggregate funciton(select sum(a), count(a)) can't be support now
5. aggregation model's table can't support hll_union/bitmap_union/count

Reply:
aggregation model's table also support hll_union/bitmap_union/count